packages/evm/src/fungible.ts

```diff
@@ -199,4 +218,16 @@ class EvmFungibleAssetTransfer extends EvmTransfer {
 
     return createTransactionRequest(transferTx);
   }
+
+  async verifyAccountBalance(): Promise<void> {
+    const resource = this.resource as EvmResource;
+    const provider = new Web3Provider(this.sourceNetworkProvider);
+    const erc20 = ERC20__factory.connect(resource.address, provider);
+    const balance = await erc20.balanceOf(this.sourceAddress);
+
+    if (BigNumber.from(this.adjustedAmount).lte(0))
```

Comment:
there is a case when the balance needs to be present in `ETH` account when fee is deducted from `BASIC` or `TWAP` fee handler however, this only checks token balance